types/src/proof/position/mod.rs

```diff
@@ -89,11 +90,13 @@ impl Position {
 
     /// What is the left node of this node? Will overflow if the node is a leaf
     pub fn left_child(self) -> Self {
+        checked_precondition!(!self.is_leaf());
```

Comment:
@zamsden ?

Reply:
You can make this an unchecked precondition if you don't want to pay the runtime cost. This method calls child, which does check the precondition. This annotation adds documentation.

Reply:
It's fine to make this a checked_precondition.  This would be a logic error which should halt execution.